src/lib/utils/search.ts

```diff
@@ -1,5 +1,5 @@
 export function filterItemsByKeyMatchingSearchQuery<T>(items: T[], key: keyof T, searchQuery: string): T[] {
-    if (!shouldSearch(searchQuery)) {
+    if (!shouldSearch(searchQuery) || (searchQuery.length === 0 && items.length > 0)) {
```

Comment:
Is this needed here? This function shouldn't be used on the library tab, only the resources tab, so if it is being used we probably need to prevent that further up.